# CBO CapTax Model Analysis
### Jaros Fork - Apple Silicon (M4) Implementation

**Purpose:** Investigate where and how CBO implements the user cost of capital elasticity in the CapTax model.

**Key Question:** CBO (2018, p.3) states: "In CBO's estimation, a 1 percent decrease in the user cost of capital translates into a 0.7 percent increase in investment."

**Repository:** https://github.com/bjaros20/captax

## Environment Setup for Apple Silicon

**Issue:** CBO's original `environment.yml` was built for Intel-based systems with Python 3.8. These exact package versions are unavailable for ARM (osx-arm64) architecture.

**Solution:** Created custom conda environment with compatible versions:

### Required Packages (vs. CBO defaults)
```bash
conda create -n CBO-CapTax python=3.9 pandas=1.3 numpy pyyaml openpyxl xlsxwriter scipy -y
conda install colorama -y  # Additional dependency discovered at runtime
```

**Key differences from CBO environment.yml:**
- Python 3.9 (vs. 3.8.5) - closest compatible version for Apple Silicon
- pandas 1.3 (vs. 1.1.3) - maintains `line_terminator` compatibility
- xlsxwriter - required for Excel output generation (missing from base)
- colorama - required for terminal color output (not in original yml)

**Result:** Model runs successfully on Apple Silicon with identical calculation logic.

In [1]:
# Import libraries
import pandas as pd
import numpy as np
import os
from pathlib import Path

# Set display options for better readability
pd.set_option('display.max_columns', None)
pd.set_option('display.width', None)
pd.set_option('display.max_rows', 50)

# Define paths
base_dir = Path('captax/data/outputs/Current-Law')
comp_dir = base_dir / 'comprehensive'
unif_dir = base_dir / 'uniformity'

print("✓ Libraries loaded")
print(f"✓ Output directories set: {comp_dir.exists()}")

✓ Libraries loaded
✓ Output directories set: True


In [4]:
# Check what files were actually created
import os

comp_path = 'captax/data/outputs/Current-Law/comprehensive/'
unif_path = 'captax/data/outputs/Current-Law/uniformity/'

print("=" * 60)
print("COMPREHENSIVE PERSPECTIVE OUTPUTS:")
print("=" * 60)
if os.path.exists(comp_path):
    for file in sorted(os.listdir(comp_path)):
        print(f"  📄 {file}")
else:
    print(f"Directory not found: {comp_path}")

print("\n" + "=" * 60)
print("UNIFORMITY PERSPECTIVE OUTPUTS:")
print("=" * 60)
if os.path.exists(unif_path):
    for file in sorted(os.listdir(unif_path)):
        print(f"  📄 {file}")
else:
    print(f"Directory not found: {unif_path}")

COMPREHENSIVE PERSPECTIVE OUTPUTS:
  📄 summary_2025_2035_Current-Law_comprehensive.csv
  📄 supplemental_table_EMTRs_Current-Law_comprehensive.xlsx
  📄 total_EMTRs_by_asset_type_legal_form_financing_2025_Current-Law_comprehensive.csv
  📄 total_EMTRs_by_industry_asset_agg_2025_Current-Law_comprehensive.csv
  📄 total_EMTRs_by_industry_legal_form_financing_2025_Current-Law_comprehensive.csv
  📄 weights_by_asset_type_legal_form_financing_2025_Current-Law_comprehensive.csv

UNIFORMITY PERSPECTIVE OUTPUTS:
  📄 summary_2025_2035_Current-Law_uniformity.csv
  📄 supplemental_table_tax_wedges_Current-Law_uniformity.xlsx
  📄 total_EMTRs_by_asset_type_legal_form_financing_2025_Current-Law_uniformity.csv
  📄 total_EMTRs_by_industry_asset_agg_2025_Current-Law_uniformity.csv
  📄 total_EMTRs_by_industry_legal_form_financing_2025_Current-Law_uniformity.csv


In [5]:
# Load the 10-year summary (2025-2035)
summary_comp = pd.read_csv(comp_path + 'summary_2025_2035_Current-Law_comprehensive.csv')
summary_unif = pd.read_csv(unif_path + 'summary_2025_2035_Current-Law_uniformity.csv')

print("COMPREHENSIVE PERSPECTIVE - Summary (2025-2035):")
display(summary_comp)

print("\nUNIFORMITY PERSPECTIVE - Summary (2025-2035):")
display(summary_unif)

COMPREHENSIVE PERSPECTIVE - Summary (2025-2035):


Key Model Metrics, by Asset Aggregate, Legal Form, and Financing Source, 2025 - 2035 (returns and EMTRs are in percent; tax wedges are in percentage points; weights are in millions of dollars)
weighting_method asset_aggs                                         legal_forms financing         year req_before_tax_returns req_after_tax_returns_savers total_tax_wedges total_EMTRs req_after_tax_returns_investors c_corp_tax_wedges c_corp_EMTRs                                            weights                                                                                                                                              
comprehensive    Nonresidential equipment                           c_corp      new_equity        2025 6.45                   5.25                         1.19             18.52       5.97                            0.48              7.39                                                 1945104.05                                                                                                                                              
                                                                                                  2026 6.62                   5.23                         1.39             20.97       5.97                            0.65              9.84                                                 1945104.05                                                                                                                                              
                                                                                                  2027 6.86                   5.24                         1.62             23.67       5.97                            0.89              13.0                                                 1945104.05                                                                                                                                              
                                                                                                  2028 6.87                   5.23                         1.64             23.81       5.97                            0.9               13.1                                                 1945104.05                                                                                                                                              
...                                                                                                                                                                                                                                                                                                   ...                                                                                                                                              
                 All equipment, structures, IPP, inventories, an... biz+ooh     typical (biz+ooh) 2031 5.07                   4.32                         0.75             14.71       NaN                             NaN               NaN                                                 94710480.63                                                                                                                                              
                                                                                                  2032 5.07                   4.33                         0.74             14.59       NaN                             NaN               NaN                                                 94710480.63                                                                                                                                              
                                                                                                  2033 5.07                   4.32                         0.74             14.7        NaN                             NaN               NaN                                                 9


UNIFORMITY PERSPECTIVE - Summary (2025-2035):


Key Model Metrics, by Asset Aggregate, Legal Form, and Financing Source, 2025 - 2035 (returns and EMTRs are in percent; tax wedges are in percentage points; weights are in millions of dollars)
weighting_method asset_aggs                                         legal_forms financing         year req_before_tax_returns req_after_tax_returns_savers total_tax_wedges total_EMTRs req_after_tax_returns_investors c_corp_tax_wedges c_corp_EMTRs                                            weights                                                                                                                                              
uniformity       Nonresidential equipment                           c_corp      new_equity        2025 5.74                   4.64                         1.1              19.11       5.33                            0.4               7.03                                                 1945104.05                                                                                                                                              
                                                                                                  2026 5.91                   4.63                         1.27             21.57       5.33                            0.58              9.74                                                 1945104.05                                                                                                                                              
                                                                                                  2027 6.13                   4.64                         1.49             24.35       5.33                            0.8               13.0                                                 1945104.05                                                                                                                                              
                                                                                                  2028 6.14                   4.64                         1.5              24.41       5.33                            0.8               13.1                                                 1945104.05                                                                                                                                              
...                                                                                                                                                                                                                                                                                                   ...                                                                                                                                              
                 All equipment, structures, IPP, inventories, an... biz+ooh     typical (biz+ooh) 2031 5.17                   4.73                         0.43             8.4         NaN                             NaN               NaN                                                 94710480.63                                                                                                                                              
                                                                                                  2032 5.16                   4.73                         0.43             8.4         NaN                             NaN               NaN                                                 94710480.63                                                                                                                                              
                                                                                                  2033 5.16                   4.73                         0.43             8.39        NaN                             NaN               NaN                                                 9

In [12]:
# Read with skiprows to handle the title row
emtr_industry = pd.read_csv(
    comp_path + 'total_EMTRs_by_industry_asset_agg_2025_Current-Law_comprehensive.csv',
    skiprows=1  # Skip the title row
)

# Rename first column
emtr_industry = emtr_industry.rename(columns={'Unnamed: 0': 'Industry'})

print("2025 EMTRs BY INDUSTRY AND ASSET TYPE")
print("=" * 80)
print(f"Shape: {emtr_industry.shape[0]} industries × {emtr_industry.shape[1]} columns")
print("\nFirst 15 industries:")
display(emtr_industry.head(15))

2025 EMTRs BY INDUSTRY AND ASSET TYPE
Shape: 75 industries × 10 columns

First 15 industries:


Industry  \
0                                               Farms   
1           Forestry, Fishing, and Related Activities   
2                              Oil and Gas Extraction   
3                         Mining (Except Oil and Gas)   
4                       Support Activities for Mining   
5                                      Electric Power   
6                            Natural Gas Distribution   
7                     Water, Sewage and Other Systems   
8                                        Construction   
9                                  Food Manufacturing   
10        Beverage and Tobacco Products Manufacturing   
11                  Textile and Textile Product Mills   
12  Apparel, Leather, and Allied Products Manufact...   
13                         Wood Product Manufacturing   
14                                Paper Manufacturing   

    Nonresidential equipment  Nonresidential structures  Residential property  \
0                       5.93                       4.11                   NaN   
1                       7.74                      22.23                   NaN   
2                      16.83                      12.84                   NaN   
3                      13.45                      16.90                   NaN   
4                      15.05                      22.35                   NaN   
5                      22.23                       9.11                   NaN   
6                      23.13                      20.44                   NaN   
7                      27.59                      21.02                   NaN   
8                       8.51                      21.86                   NaN   
9                      13.13                      28.08                   NaN   
10                     10.46                      25.61                   NaN   
11                     14.76                      28.63                   NaN   
12                     12.23                      27.54                   NaN   
13                     15.42                      28.43                   NaN   
14                     14.63                      27.57                   NaN   

    R&D and own-account software  Other intellectual property products  \
0                            NaN                                 16.64   
1                            NaN                                 17.45   
2                           1.87                                 22.58   
3                          -3.47                                 12.78   
4                           3.08                                 22.66   
5                          -2.94                                 21.33   
6                           0.39                                 23.15   
7                          -1.07                                 21.97   
8                          -2.02                                 19.91   
9                          -6.40                                 22.36   
10                        -15.91                                 19.17   
11                         -4.44                                 22.84   
12                         -9.00                                 21.41   
13                         -7.07                                 22.17   
14                         -7.37                                 21.94   

    Inventories  All equipment, structures, IPP, and inventories   Land  \
0         28.98                                            10.29  27.48   
1         29.03                                            14.80  27.46   
2         29.89                                            12.85  27.62   
3         30.30                                            16.74  27.87   
4         30.44                                            19.15  28.14   
5         28.74                                            13.53  25.58   
6         30.03                                            20.98  27.23   
7         29.16                                   

In [13]:
# Focus on the "All equipment, structures, IPP, inventories, and land" column
overall_col = 'All equipment, structures, IPP, inventories, and land'

# Sort industries by overall EMTR
emtr_sorted = emtr_industry[['Industry', overall_col]].copy()
emtr_sorted = emtr_sorted.sort_values(by=overall_col, ascending=False).dropna()

print("TOP 15 INDUSTRIES BY OVERALL EMTR (2025):")
print("=" * 60)
display(emtr_sorted.head(15))

print("\nBOTTOM 15 INDUSTRIES BY OVERALL EMTR (2025):")
print("=" * 60)
display(emtr_sorted.tail(15))

# Get overall economy EMTR
overall_emtr = emtr_industry[emtr_industry['Industry'] == 'All Industries'][overall_col].values[0]
print(f"\n📊 OVERALL ECONOMY EMTR: {overall_emtr:.2f}%")

TOP 15 INDUSTRIES BY OVERALL EMTR (2025):


Industry  \
31                         Wholesale Nondurable Goods   
32                    Motor Vehicle and Parts Dealers   
30                            Wholesale Durable Goods   
34                         General Merchandise Stores   
13                         Wood Product Manufacturing   
45      Motion Picture and Sound Recording Industries   
33                           Food and Beverage Stores   
11                  Textile and Textile Product Mills   
55                                        Real Estate   
43                            Warehousing and Storage   
63                               Educational Services   
12  Apparel, Leather, and Allied Products Manufact...   
27       Furniture and Related Products Manufacturing   
16          Petroleum and Coal Products Manufacturing   
0                                               Farms   

    All equipment, structures, IPP, inventories, and land  
31                                              27.27      
32                                              26.82      
30                                              25.89      
34                                              25.63      
13                                              25.06      
45                                              24.98      
33                                              24.93      
11                                              24.79      
55                                              24.78      
43                                              24.64      
63                                              24.41      
12                                              24.03      
27                                              23.74      
16                                              23.61      
0                                               23.56


BOTTOM 15 INDUSTRIES BY OVERALL EMTR (2025):


Industry  \
48                   Depository Credit Intermediation   
36                                 Air Transportation   
56  Rental and Leasing (Including Nonfinancial Int...   
65                                          Hospitals   
5                                      Electric Power   
59  Other Professional, Scientific, and Technical ...   
37                                Rail Transportation   
50                Nondepository Credit Intermediation   
41                            Pipeline Transportation   
17                             Chemical Manufacturing   
22                            Machinery Manufacturing   
25              Motor Vehicle and Parts Manufacturing   
44            Publishing Industries (Except Internet)   
23      Computer and Electronic Product Manufacturing   
24  Electrical Equipment, Appliances, and Componen...   

    All equipment, structures, IPP, inventories, and land  
48                                              15.43      
36                                              14.84      
56                                              14.18      
65                                              14.12      
5                                               13.95      
59                                              13.58      
37                                              13.42      
50                                              13.13      
41                                              12.76      
17                                              12.11      
22                                               8.56      
25                                               8.47      
44                                               5.58      
23                                              -3.40      
24                                            -133.86


📊 OVERALL ECONOMY EMTR: 20.38%


In [14]:
# Look at aggregate row only
all_industries_row = emtr_industry[emtr_industry['Industry'] == 'All Industries'].iloc[0]

print("AGGREGATE EMTRs BY ASSET TYPE (2025):")
print("=" * 60)

asset_data = [
    ('Nonresidential Equipment', all_industries_row['Nonresidential equipment']),
    ('Nonresidential Structures', all_industries_row['Nonresidential structures']),
    ('R&D and Software', all_industries_row['R&D and own-account software']),
    ('Other IP Products', all_industries_row['Other intellectual property products']),
    ('Inventories', all_industries_row['Inventories']),
    ('Land', all_industries_row['Land']),
    ('Overall (All Assets)', all_industries_row['All equipment, structures, IPP, inventories, and land'])
]

for asset, emtr in asset_data:
    print(f"  {asset:.<35} {emtr:>8.2f}%")

AGGREGATE EMTRs BY ASSET TYPE (2025):
  Nonresidential Equipment...........    12.12%
  Nonresidential Structures..........    20.79%
  R&D and Software...................   -16.89%
  Other IP Products..................    22.46%
  Inventories........................    29.44%
  Land...............................    26.73%
  Overall (All Assets)...............    20.38%


In [16]:
# Let's look at the context of "0.7" in these files
import pandas as pd

print("INVESTIGATING '0.7' OCCURRENCES")
print("=" * 80)

# Check the main policy parameters file
print("\n1. POLICY PARAMETERS FILE:")
policy_params = pd.read_csv('captax/data/inputs/policy_parameters/policy_parameters_Current-Law_comprehensive.csv')
print(f"Shape: {policy_params.shape}")
print("Looking for columns or values with 0.7...")

# Search for 0.7 in the dataframe
for col in policy_params.columns:
    if policy_params[col].astype(str).str.contains('0.7', na=False).any():
        print(f"\nColumn '{col}' contains 0.7:")
        matches = policy_params[policy_params[col].astype(str).str.contains('0.7', na=False)]
        display(matches[[col]].head())

print("\n" + "=" * 80)
print("\n2. SECTION 199A ADJUSTMENTS:")
sec_199a = pd.read_csv('captax/data/inputs/policy_parameters/tax_rate_adjustments/sec_199A_adjustments_CLaw_temp.csv')
print("First few rows:")
display(sec_199a.head())

print("\n" + "=" * 80)
print("\n3. ECONOMIC DEPRECIATION RATES:")
econ_dep = pd.read_csv('captax/data/inputs/environment_parameters/economic_depreciation.csv')
print("Checking for 0.7 in depreciation rates...")
econ_dep_str = econ_dep.astype(str)
if (econ_dep_str == '0.7').any().any():
    print("Found 0.7 values:")
    display(econ_dep[econ_dep.astype(str).isin(['0.7']).any(axis=1)])
else:
    print("No exact 0.7 values in economic depreciation")

INVESTIGATING '0.7' OCCURRENCES

1. POLICY PARAMETERS FILE:
Shape: (11, 71)
Looking for columns or values with 0.7...

Column 'year' contains 0.7:


year
2  2027


Column 'pass_thru_eligibility_below_thresholds' contains 0.7:


pass_thru_eligibility_below_thresholds
0                                  0.7712


Column 'prop_tax_deductible_share' contains 0.7:


prop_tax_deductible_share
1                     0.7523
2                     0.7680
3                     0.7683
4                     0.7705
5                     0.7704


Column 'itc_rates' contains 0.7:


itc_rates
2  CLaw_2027


Column 'ptc_rates' contains 0.7:


ptc_rates
2  CLaw_2027


Column 'ooh_debt_account_category_share_taxable' contains 0.7:


ooh_debt_account_category_share_taxable
6                                   0.8097


Column 'ooh_debt_account_category_share_deferred' contains 0.7:


ooh_debt_account_category_share_deferred
0                                    0.0573
1                                    0.0573
2                                    0.0573
3                                    0.0573
4                                    0.0573



2. SECTION 199A ADJUSTMENTS:
First few rows:


Section 199A parameters by industry  \
0                            Default values   
1                                     Farms   
2  Forestry,_Fishing,_and_Related_Activites   
3                    Oil_and_Gas_Extraction   
4               Mining_(Except_Oil_and_Gas)   

   Section 199A deduction eligibilty  Section 199A deduction rate  
0                             0.7826                          0.2  
1                             0.7826                          0.2  
2                             0.7826                          0.2  
3                             0.7826                          0.2  
4                             0.7826                          0.2



3. ECONOMIC DEPRECIATION RATES:
Checking for 0.7 in depreciation rates...
No exact 0.7 values in economic depreciation


## 🔍 Analysis of "0.7" Values Found in Input Files

**All "0.7" occurrences are TAX POLICY PARAMETERS, not elasticity:**

### 1. Section 199A (QBI Deduction) Parameters:
- **pass_thru_eligibility_below_thresholds: 0.77** 
  - Proportion of pass-through income eligible for QBI deduction below income thresholds
- **Section 199A deduction rate: 0.2** (20%)
  - The actual QBI deduction percentage

### 2. Property Tax Deduction:
- **prop_tax_deductible_share: 0.75-0.77**
  - Share of property taxes that are deductible

### 3. Other occurrences:
- Year **2027** (contains "0.7" as substring)
- Various account category shares around 0.7

### ✅ CONFIRMED: No Investment Elasticity in CapTax

**None of these 0.7 values represent the investment response elasticity.** They are all tax code parameters that define:
- How much of pass-through income qualifies for deductions
- What share of property taxes can be deducted  
- Other tax calculation rules

**The CBO (2018) elasticity of 0.7 is NOT in CapTax's inputs or code.**

## Final Conclusion: Where is the 0.7 Elasticity?

### Search Results Summary:
✅ **Python code**: No "elasticity", "0.7 elasticity", or "investment response"  
✅ **Input parameter files**: Found "0.7" in 11 files  
✅ **Verification**: All "0.7" values are tax policy parameters, not elasticity

### What CapTax Does:
CapTax calculates **effective marginal tax rates (EMTRs)** which represent the change in **user cost of capital**:
- User cost = (r + δ - π + τ) / (1 - τ)
- Where τ = EMTR

### Where the Elasticity Lives:
The elasticity is in **CBO's downstream macroeconomic model**, which:
1. Takes CapTax's EMTR outputs
2. Converts EMTR changes to % changes in user cost
3. Applies: **ΔInvestment = -0.7 × ΔUserCost**
4. Feeds into capital accumulation and GDP projections

### For Documentation:
> "CapTax computes effective marginal tax rates by industry, asset, legal form, and financing source. It does not contain or apply the investment elasticity. The 0.7 elasticity cited in CBO (2018) is applied in CBO's separate macroeconomic model, which uses CapTax outputs to estimate impacts on investment, capital stock, and GDP."

##  The Source of the 0.7 Elasticity
### From CBO (2018) Report on TCJA Macroeconomic Effects:

> "In CBO's estimation, a 1 percent decrease in the user cost of capital translates into a 0.7 percent increase in investment."

**Footnote 5 reveals:**
> "That value is CBO's own estimate, and it is near the middle of the range of values estimated by other researchers. For a review of that literature, see Kevin A. Hassett and R. Glenn Hubbard, 'Tax Policy and Business Investment,' in Handbook of Public Economics, vol. 3 (2002)."

### Key Findings:

**The 0.7 is NOT calculated - it's a PARAMETER:**
- CBO selected 0.7 based on empirical literature review
- It represents the middle of the range from academic studies
- It's a **calibrated assumption**, not an endogenously derived value

**The Workflow:**
1. **CapTax calculates** → User cost changes (% change in EMTR)
2. **CBO's macro model** → Applies the 0.7 elasticity parameter
3. **Result** → Investment impact: ΔI = -0.7 × Δ(user cost)

**Quote from CBO (2018):**
> "CBO calculated the act's changes in the user costs of capital faced by C corporations and by pass-through businesses for 32 types of equipment, 23 types of nonresidential structures, 3 types of intellectual property products, 3 types of residential capital, and inventories."

This is exactly what CapTax does! It computes user cost changes by detailed categories, then CBO's separate model applies the elasticity.

##  Elasticity Source Documentation

**Parameter:** 0.7 investment elasticity  
**Source:** CBO's own estimate (2018)  
**Basis:** Middle of range from academic literature (Hassett & Hubbard, 2002)  
**Location:** Applied in CBO's macroeconomic model, **NOT in CapTax**  
**Usage:** ΔInvestment = -0.7 × Δ(User Cost of Capital)

### CapTax's Role:
- ✅ Calculates effective marginal tax rates (EMTRs) 
- ✅ Computes user cost changes by:
  - 32 equipment types
  - 23 structure types
  - 3 IP types
  - 3 residential types
  - Inventories
- ✅ Separates by C-corp vs pass-through
- ✅ Provides **inputs** to macro model which applies the 0.7 elasticity

### CBO's Workflow:
```
CapTax → User Cost Changes → [Macro Model] → Apply 0.7 Elasticity → Investment Impacts
```

### Reference:
Congressional Budget Office (2018). "Key Methods CBO Used to Estimate the Macroeconomic Effects of the 2017 Tax Act"  
https://www.cbo.gov/system/files/115th-congress-2017-2018/reports/keymethodsthatcbousedtoestimatethemacroeconomiceffectsofthe2017taxact.pdf

### Conclusion:
**CBO's macroeconomic model (which applies the elasticity) is not on GitHub.** Only CapTax (the tax calculator) is published on GitHub.